### PR TITLE
Fixed some MSVC errors

### PR DESCRIPTION
**Pull request by NotRequiem**
The code was modified to handle wide character strings instead of narrow character strings using MultiByteToWideChar. The original code used narrow character strings (LPCSTR or const char*) that were causing compilation issues on MSVC VS 2022

The code was also compiled and tested after the changes to ensure the modifications are ok

> 1. Compilation errors in the original code:

> 2. Compilation status in the modified code:

### CODE REVIEW

**Comment by kernelwernel** [comment]
I appreciate it, but can you tell me what compiler flags you're using? The compiler flags for `cl` with the cmake build is `/W4 /std:c++20 /EHsc`, but yours seems to be different.

File: src/vmaware.hpp
Comment (kernelwernel) [inline src/vmaware.hpp:1773]: I don't really see the point of the indentation, but it really doesn't matter. I'll update that in a future commit, don't worry about it.

File: src/vmaware.hpp
Comment (kernelwernel) [inline src/vmaware.hpp:2094]: same here

**Review by kernelwernel — APPROVED** [review]
overall, looks good to me :)

**Comment by NotRequiem** [comment]
> I appreciate it, but can you tell me what compiler flags you're using? The compiler flags for `cl` with the cmake build is `/W4 /std:c++20 /EHsc`, but yours seems to be different.

im just using /std:c++20 with MSVC, without any other flag

**Comment by kernelwernel** [comment]
I just want your input before I merge this, if you have any extra suggestions/changes to request then don't hesitate. If it looks good to you then let me know.

**Comment by kernelwernel** [comment]
I really appreciate the PR btw, thanks for being the first contributor to this project :)

**Pull request merged**
